linkerd/app/outbound/src/tcp/logical/router.rs

```diff
@@ -133,6 +180,15 @@ impl<Req, S: Service<Req> + Clone> Service<Req> for Router<S> {
     type Future = S::Future;
 
     fn poll_ready(&mut self, cx: &mut Context<'_>) -> Poll<Result<(), S::Error>> {
+        // does the current distribution need to be updated?
+        if self
+            .rx
+            .has_changed()
+            .expect("router background task terminated unexpectedly!")
+        {
+            self.current = self.rx.borrow().clone();
+        }
+
```

Comment:
This is definitely better; though, I think it means we should remove the update in `call`?